pyingest/parsers/gcncirc.py

```diff
@@ -45,33 +62,60 @@ def split_authors_abstract(self):
         while body[0] != '' and ':' not in body[0]:
             auths.append(body.pop(0).strip())
         auths.append(body.pop(0).strip())
+
         auth_string = ' '.join(auths)
-        auth_array = auth_string.split(',')
-        print "AUTHORS: ",auth_array
-        for a in auth_array:
-            aff = re.search(r'\((.*?)\)',a).group(1)
-            # aff = a[a.find("(")+1:a.find(")")]
-            # print "auth:",a
-#           print "lol aff:",aff
+
+        auth_string = re.sub(r'\s+\((.*?)\)\s+', ',', auth_string)
+        auth_string = re.sub(r'[ ,]and\s', ',', auth_string)
+        auth_string = re.sub(r'on behalf of',',',auth_string)
+        auth_string = re.sub(r'reports?', ',', auth_string)
+        auth_string = re.sub(r'\s?:','', auth_string)
+        auth_string = re.sub(r',\s+,',',', auth_string)
+        
+        auth_array = [s.strip() for s in auth_string.split(',')]
+        auth_array = list(filter(lambda a: len(a) > 3, auth_array))
+        auth_string = '; '.join(auth_array)
+        auth_mod = AuthorNames()
+        self.data_dict['authors'] = auth_mod.parse(auth_string)
+        
+        
 
     def parse(self):
 
         self.data_dict = {}
+        # Start by looking at the Circular line by line...
         try:
             gdata = self.raw.split('\n')
 
+        # Header is fixed format and five lines long...
             head = gdata[0:5]
             for l in head:
                 lfix = l.replace(' ','\t',1)
                 lparts = lfix.split('\t')
                 self.data_dict[head_dict[lparts[0]]] = lparts[1].strip()
+            # Now you need to split the authors from the abstract.
+            # This should work in *most* cases, maybe not all,
+            # especially from older (pre-2016) Circulars
             self.data_dict['abstract'] = gdata[5:]
             self.split_authors_abstract()
-#           body = [x for x in gdata[5:] if x != '']
-#           self.data_dict['abstract'] = '<pre>' + '\n'.join(body) + '<\\pre>'
+            # Authors and abstract content should now be defined
+            
+            # If you want to try and keep fixed formatting
+            # (e.g. for tables), use '\n' for the join character
```

Comment:
Agree it would be nice to keep tables somewhat decent, but that looks like a difficult task so I'd keep it for a later time.  There is other formatting that could be done better, e.g. use paragraph breaks maybe for separating a reference section (looks like the example GCN had two empty lines).  I wouldn't do it for every single empty line since they are very liberal with those.

Reply:
I'd like to defer this to a subsequent PR.